src/arrayMethodSort.js

```diff
@@ -6,6 +6,30 @@
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
     // write code here
+    const array = this;
+    const length = array.length;
+
+    for (let i = 0; i < length - 1; i++) {
+      for (let j = 0; j < length - 1 - i; j++) {
+        if (compareFunction && typeof compareFunction === 'function') {
```

Comment:
compareFunction should always exist, so this checking is redundant. Just add a default value for the compareFunction in arguments 
```
function(compareFunction = defaultCompareFn) { 
 ...